cmd/commands/upgrade.go

```diff
@@ -0,0 +1,203 @@
+/*
+ *    Copyright 2024 Han Li and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package commands
+
+import (
+	"path/filepath"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"github.com/urfave/cli/v2"
+	"archive/tar"
+	"compress/gzip"
+)
+
+var Upgrade = &cli.Command{
+    Name:   "upgrade",
+    Usage:  "Upgrade vfox to the latest version.",
+    Action: upgradeCmd,
+}
+
+type ReleaseInfo struct {
+	TagName string `json:"tag_name"`
+}
+
+func getLatestReleaseInfo(apiURL string) (ReleaseInfo, error) {
+	var releaseInfo ReleaseInfo
+
+	resp, err := http.Get(apiURL)
+	if err != nil {
+		return releaseInfo, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return releaseInfo, err
+	}
+
+	err = json.Unmarshal(body, &releaseInfo)
+	return releaseInfo, err
+}
+
+func constructBinaryName(tagName string) string {
+	osType := strings.ToLower(runtime.GOOS)
+	if osType == "darwin" {
+		osType = "macos"
+	}
+
+	archType := runtime.GOARCH
+	if archType == "arm64" {
+		archType = "aarch64"
+	}
+
+	fileName := fmt.Sprintf("vfox_%s_%s_%s.tar.gz", tagName[1:], osType, archType) 
+	return fileName
+}
+
+// constructDiffURL constructs a GitHub diff URL between two version tags
+func constructDiffURL(oldTag, newTag string) string {
+    return fmt.Sprintf("https://github.com/version-fox/vfox/compare/%s...%s", oldTag, newTag)
+}
+
+
+func getUrls(apiURL string,currVersion string) (string, string) {
+	version:= currVersion;
+    releaseInfo, err := getLatestReleaseInfo(apiURL)
+    if err != nil {
+        fmt.Println("Error fetching release info:", err)
+        return "", ""
+    }
+	fileName := constructBinaryName(releaseInfo.TagName)
+	binURL := fmt.Sprintf("https://github.com/version-fox/vfox/releases/download/%s/%s", releaseInfo.TagName, fileName)
+	diffURL := constructDiffURL(version,releaseInfo.TagName)
+    return binURL, diffURL
+}
+
+func downloadFile(filepath string, url string) error {
+    out, err := os.Create(filepath)
+    if err != nil {
+        return err
+    }
+    defer out.Close()
+    resp, err := http.Get(url)
+    if err != nil {
+        return err
+    }
+    defer resp.Body.Close()
+    _, err = io.Copy(out, resp.Body)
+    return err
+}
+
+func extractTarGz(gzipFilePath, destDir string) error {
```

Comment:
There is a decompression tool in the project. Please use that tool to replace this function. There is no need to reimplement it. 

> ⚠️ NOTE: Slightly different behavior

https://github.com/version-fox/vfox/blob/d9dd9937c979d0202beccf20cca4c4191bee68e5/internal/util/decompressor.go#L231